pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java

```diff
@@ -167,8 +168,10 @@ public HLRealtimeSegmentDataManager(final RealtimeSegmentZKMetadata realtimeSegm
     // create and init stream level consumer
     _streamConsumerFactory = StreamConsumerFactoryProvider.create(_streamConfig);
     String clientId = HLRealtimeSegmentDataManager.class.getSimpleName() + "-" + _streamConfig.getTopicName();
+    Set<String> sourceFields = SchemaFieldExtractorUtils.extract(schema);
```

Comment:
Shouldnt this be _destination_ fields instead of source fields? Or, am I missing something?

Reply:
Renamed the method to `extractSourceFields`. It should be source fields - we get the source fields upfront and pass it to the decoder/reader.